src/components/EmojiSuggestions.vue

```diff
@@ -0,0 +1,176 @@
+<!--
+  - @copyright Copyright (c) 2021 Jonas Meurer <jonas@freesources.org>
+  -
+  - @author Jonas Meurer <jonas@freesources.org>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="suggestion-list"
+		:class="{ 'is-active': showSuggestions }"
+		:style="position()">
+		<template v-if="hasResults">
+			<div v-for="(emojiObject, index) in filteredEmojis"
+				:key="emojiObject.id"
+				class="suggestion-list__item"
+				:class="{ 'is-selected': navigatedEmojiIndex === index }"
+				@click="selectEmoji(emojiObject)">
+				{{ emojiObject.value }} {{ emojiObject.short_name }}
+			</div>
+		</template>
+		<div v-else class="suggestion-list__item is-empty">
+			{{ t('text', 'No emojis found') }}
+		</div>
+	</div>
+</template>
+
+<script>
+
+export default {
+	name: 'EmojiSuggestions',
+	props: {
+		emojiQuery: {
+			type: String,
+			required: true,
+		},
+		emojiRange: {
+			type: Object,
+			required: true,
+		},
+		filteredEmojis: {
+			type: Array,
+			required: true,
+		},
+		navigatedEmojiIndex: {
+			type: Number,
+			required: true,
+		},
+		emojiRect: {
+			type: Object,
+			required: true,
+		},
+		insertEmoji: {
+			type: Function,
+			required: true,
+		},
+	},
+	computed: {
+		hasResults() {
+			return this.filteredEmojis.length
+		},
+
+		showSuggestions() {
+			return this.emojiQuery || this.hasResults
+		},
```

Comment:
For readability and to make sure we manipulate booleans here.
```suggestion
		hasResults() {
			return this.filteredEmojis.length > 0
		},

		hasEmojiQuery() {
			return this.emojiQuery !== null && this.emojiQuery !== ''
		},
		
		showSuggestions() {
			return this.hasEmojiQuery || this.hasResults
		},		
```

Reply:
Full ack. Much cleaner that way :)